fix/docs: fix `get_valid_oeis_id`, fix shell script, add shell script, add some docs

liammulh opened the pull request:
This PR:

- Fixes the `get_valid_oeis_id` function. When you tried to set `valid_id[0] = first_character.upper()`, you'd get `TypeError: 'str' object does not support item assignment`.
- Fixes a typo in a shell script.
- Adds a shell script that you can run to add the `numberscope` systemd service.
- Adds some comments, and a note to the `server/README.md` file.

gwhitney commented on flaskr/nscope/views.py:
```diff
@@ -205,7 +205,8 @@ def get_valid_oeis_id(oeis_id):
                 # https://github.com/numberscope/backscope/issues/57.
                 print('info: first character in oeis_id is lowercase')
                 print('info: making first character in oeis_id uppercase')
```
Not clear to me that (even if logging is working) we need to log instances of requests with lower case IDs -- probably a common typo, and who is ever going to be interested in those log messages? Likely better to reserve the log for more significant information (plus we don't want it to grow too much, then it's hard to find the stuff that matters among everything else).

liammulh replied:
I imagine we can configure logging levels, e.g. info, debug, warn, verbose, etc. When we have proper logging, these could be verbose.

In my experience logs like these can be helpful if there's a bug and you need to look through the logs for clues as to what might be going wrong.

While I don't care _that much_, I feel like having more information in the logs isn't a bad thing if you can configure levels. If there's only one level, then I absolutely agree that this doesn't rise to the level of being log-worthy.

I'll add a comment in the code.

gwhitney replied:
OK, I would be fine with this being a verbose level log message that would (generally) be off in production. The print can serve as a placeholder for that for now.

gwhitney commented on flaskr/nscope/views.py:
```diff
@@ -205,7 +205,8 @@ def get_valid_oeis_id(oeis_id):
-                valid_id[0] = first_character.upper()
+                valid_id = first_character.upper()
+                valid_id += oeis_id.partition(first_character)[2]
     else:
```
The idiomatic python for this operation is `valid_id = first_character.upper() + valid_id[1:]`

liammulh replied:
Okay, thanks! I will change it.

gwhitney reviewed (commented):
Just a couple of small comments, otherwise looks good. Sorry I didn't catch that typo before in the create_symlinks script. Let me know when you're done and I will merge.

liammulh commented:
@gwhitney, I addressed your comments in https://github.com/numberscope/backscope/pull/59/commits/eef63e7e2256279dbd5557619af504f9535841c3 and manually performed regression tests to ensure the `get_valid_oeis_id` function is still working as intended.

gwhitney commented:
OK, merging. Let's get the repo and production in sync, and then keep production on main insofar as possible. Thanks!
